src/tests/panel.test.ts

```diff
@@ -79,77 +76,59 @@ describe('_getPanelsFromPanelSetIDController', () => {
-        const response = await _createPanelController(sequelizeMock)('image.png', 3, 1);
-        expect(response).toBe(UpdatePanelObj);
+    test('if panelSet.panels is not null, a panel should be created', async() => {
+        const panelSet = { panels: ['a', 'b'] };
+        (getPanelSetByID as jest.Mock).mockReturnValue(() => Promise.resolve(panelSet));
+        (panelService.createPanel as jest.Mock).mockReturnValue(() => Promise.resolve(panelObj));
+        const response = await panel._createPanelController(sequelizeMock)('', 0);
+        expect(response).toBe(panelObj);
     });
 
-    // if existing panel isn't found  create, return json object
-    test('if panel doesnt exist, create it and return the created object from the createPanel', async () =>{
-        (getPanelSetByID as jest.Mock).mockReturnValue(() => Promise.resolve(getPanelSetObj));
-        (panelService.getPanelBasedOnPanelSetAndIndex as jest.Mock).mockReturnValue(() => Promise.resolve(undefined));
-        (panelService.createPanel as jest.Mock).mockReturnValue(() => Promise.resolve(CreatePanelObj));
-        const response = await _createPanelController(sequelizeMock)('image.png', 3, 1);
-        expect(response).toBe(CreatePanelObj);
+    test('if panelSet.panels is null, a panel should still be created', async() => {
+        const panelSet = {};
+        (getPanelSetByID as jest.Mock).mockReturnValue(() => Promise.resolve(panelSet));
+        (panelService.createPanel as jest.Mock).mockReturnValue(() => Promise.resolve(panelObj));
+        const response = await panel._createPanelController(sequelizeMock)('', 0);
+        expect(response).toBe(panelObj);
     });
+});
 
-    // error creating
-    test('if something throws an error in one of the services', async () =>{
-        (getPanelSetByID as jest.Mock).mockReturnValue(() => Promise.resolve(getPanelSetObj));
-        (panelService.getPanelBasedOnPanelSetAndIndex as jest.Mock).mockReturnValue(() => Promise.resolve(undefined));
-        (panelService.createPanel as jest.Mock).mockReturnValue(() => Promise.resolve(new Error('Error creating panel')));
-        const response = await _createPanelController(sequelizeMock)('', 3, -1);
+describe('_updatePanelController', () => {
+    test('if panel does not exist, throw an error', async () => {
+        (panelService.getPanel as jest.Mock).mockReturnValue(() => Promise.resolve(null));
+        const response = await panel._updatePanelController(sequelizeMock)(0, '');
+        expect(response).toBeInstanceOf(Error);
+    });
+    test('if panel does exist, return the updated panel', async () => {
+        const panelObj = { a: 'abcd' };
+        (panelService.getPanel as jest.Mock).mockReturnValue(() => Promise.resolve('a'));
+        (panelService.updatePanel as jest.Mock).mockResolvedValue(panelObj);
+        const response = await panel._updatePanelController(sequelizeMock)(0, '');
+        expect(response).toBe(panelObj);
```

Comment:
This test is failing. It says `panel` is undefined. The `_updatePanelController` no longer calls `getPanel` so the three tests here are inaccurate to the actual function, which is likely the source of the error.